### ISSUE TITLE
`daySelectedFirst` not applied to ranges (only globally)

### ISSUE THREAD

**Issue by johanrd** [uvarov-frontend/vanilla-calendar-pro#313]
## Issue:
When having multiple selected periods, I'd expect the first and last of each range to be rounded:

| Actual  |      Expected      |
|----------|:-------------:|
| <img width="294" alt="Screenshot 2024-10-22 at 15 03 24" src="https://github.com/user-attachments/assets/a66d76cc-c7cb-433d-be33-93af187a3b33"> | <img width="293" alt="Screenshot 2024-10-22 at 15 04 39" src="https://github.com/user-attachments/assets/a7f0f088-8998-41b8-9a7b-52dda2297f02"> |

## Proposed solution 1:
Apply `daySelectedFirst` and `daySelectedLast` to all ranges, not only the first and last (globally)

## Proposed solution 2:
Add separate `daySelectedFirstInRange` and `daySelectedLastInRange` settings

## Current workaround:
For styling purposes, you could always do some css-fighting like this, but it is not very elegant:

```css
.vanilla-calendar-day_selected-first .vanilla-calendar-day__btn_selected,
.vanilla-calendar-day:not(.vanilla-calendar-day_selected-intermediate):not(.vanilla-calendar-day_selected-first) + .vanilla-calendar-day_selected-intermediate .vanilla-calendar-day__btn_selected
 {
  border-top-left-radius: .5rem !important;
  border-bottom-left-radius: .5rem !important;
}

.vanilla-calendar-day_selected-first:has(+ .vanilla-calendar-day:not(.vanilla-calendar-day_selected-intermediate)) .vanilla-calendar-day__btn_selected,
.vanilla-calendar-day_selected-intermediate:has(+ .vanilla-calendar-day:not(.vanilla-calendar-day_selected-intermediate)) .vanilla-calendar-day__btn_selected
{
  border-top-right-radius: .5rem !important;
  border-bottom-right-radius: .5rem !important;
}
```

**Comment by uvarov-frontend** [comment]
Oh, it would be great to use the second option, since implementing such decorative things through js is not optimal. We can add a catch to support the new `:has`, this will work in modern browsers, thanks for the idea.

**Comment by uvarov-frontend** [comment]
Fixed in v3.0.0 #293
